Description: Fix crash in pair_content()
 Fix for a crash in pair_content() cherry-picked from upstream
 patchlevel 20180519: add check in pair_content() to handle the case
 where caller asks for an uninitialized pair (Debian #898658).
Author: Sven Joachim <svenjoac@gmx.de>
Forwarded: not-needed
Bug-Debian: https://bugs.debian.org/898658
Last-Update: 2018-05-20

---
 ncurses/base/lib_color.c |    7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

--- a/ncurses/base/lib_color.c
+++ b/ncurses/base/lib_color.c
@@ -938,9 +938,12 @@ _nc_pair_content(SCREEN *sp, int pair, i
     if (!ValidPair(sp, pair)) {
 	result = ERR;
     } else {
-	int fg = FORE_OF(sp->_color_pairs[pair]);
-	int bg = BACK_OF(sp->_color_pairs[pair]);
+	int fg;
+	int bg;
 
+	_nc_reserve_pairs(sp, pair);
+	fg = FORE_OF(sp->_color_pairs[pair]);
+	bg = BACK_OF(sp->_color_pairs[pair]);
 #if NCURSES_EXT_FUNCS
 	if (isDefaultColor(fg))
 	    fg = -1;
